src/DotVVM.Framework/Compilation/ExtensionMethodsCache.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace DotVVM.Framework.Compilation
+{
+    public class ExtensionMethodsCache
+    {
+        private readonly CompiledAssemblyCache assemblyCache;
+        private readonly ConcurrentDictionary<string, List<MethodInfo>> methodsCache;
```

Comment:
I'd prefer to cache ImmutableArray to prevent accidental cache poisoning by changing the returned List.